### PR TITLE
Adding RFC links for standardised HttpHeaderNames constants in javadoc

### PR DESCRIPTION
**Pull request by shekhar-rajak**
**Motivation**

Enhance javadoc for HTTP header name constants in HttpHeaderNames class with links to the related RFC sections.

**Modifications**

Adding `@see` like:

```
 @see <a href="https://tools.ietf.org/html/rfc7230#section-6.1">RFC7230, section 6.1</a>
```

in `HttpHeaderNames.java` file for constants.

**Result**

 javadoc for HTTP header name constants will be updated with related RFC sections.

Fixes: https://github.com/apple/servicetalk/issues/885

### CODE REVIEW

**Comment by NiteshKant** [comment]
@servicetalk-bot test this please

File: servicetalk-http-api/src/main/java/io/servicetalk/http/api/HttpHeaderNames.java [hunk 1]
```diff
@@ -180,70 +233,104 @@
     public static final CharSequence FORWARDED = newAsciiString("forwarded");
```
Comment (idelpivnitskiy) [inline servicetalk-http-api/src/main/java/io/servicetalk/http/api/HttpHeaderNames.java]: While you are here, could you align javadoc for this header to follow a consistent format:

```
{@code "forwarded"} is a header field that ...
...

@see <a href="https://tools.ietf.org/html/rfc7239#section-4">RFC7239, section 4</a>
```

Please, also adjust `X_FORWARDED_FOR`, `X_FORWARDED_HOST`, and `X_FORWARDED_PROTO` constant. to use the same format.

File: servicetalk-http-api/src/main/java/io/servicetalk/http/api/HttpHeaderNames.java [hunk 2]
```diff
@@ -264,66 +351,98 @@
...
     /**
      * {@code "via"}
+     *
+     * @see <a href="https://tools.ietf.org/html/rfc7230#section-5.7.1">RFC7230, section 5.7.1</a>
      */
     public static final CharSequence VIA = newAsciiString("via");
     /**
      * {@code "warning"}
+     *
+     * @see <a href="https://tools.ietf.org/html/rfc7234#section-5.5">RFC7234, section 5.5 </a>
```
Comment (idelpivnitskiy) [inline servicetalk-http-api/src/main/java/io/servicetalk/http/api/HttpHeaderNames.java]: Please, remove unnecessary space before closing tag `5.5 _</a>`

File: servicetalk-http-api/src/main/java/io/servicetalk/http/api/HttpHeaderNames.java [hunk 3]
```diff
@@ -340,6 +459,8 @@
     public static final CharSequence WEBSOCKET_PROTOCOL = newAsciiString("websocket-protocol");
     /**
      * {@code "www-authenticate"}
+     *
+     * @see <a href="https://tools.ietf.org/html/rfc#section-">RFC, section </a>
```
Comment (idelpivnitskiy) [inline servicetalk-http-api/src/main/java/io/servicetalk/http/api/HttpHeaderNames.java]: Looks like a missed link to https://tools.ietf.org/html/rfc7235#section-4.1

**Comment by idelpivnitskiy** [comment]
@shekhar-rajak I will suggest writing a verbal description in `Motivation` section of the commit/PR message. Something like:

Enhance javadoc for HTTP header name constants in `HttpHeaderNames` class with links to the related RFC sections.

At the end of the full message (after `Results` section), you can refer to the issue id to let github automatically close it when PR is merged.

**Comment by shekhar-rajak** [comment]
Thanks @idelpivnitskiy I will keep this in mind. I have edited the PR description.

**Comment by idelpivnitskiy** [comment]
@servicetalk-bot test this please

**Comment by idelpivnitskiy** [comment]
Gradle crashed :(

@servicetalk-bot test this please

**Pull request merged**